### ISSUE TITLE
dat://archive.better.fyi crashes every time

### ISSUE THREAD

**Issue by vuori** [beakerbrowser/beaker#1064]
Operation System: Windows 10 1803
Beaker version: Beaker v0.8.0-prerelease.6

Navigating to `dat://archive.better.fyi` crashes the browser deterministically on Windows 10: app becomes unresponsive immediately, fans run full blast for a while and browser exits. Other platforms not tested.

Nothing is logged to  `dat.log` or `debug.log`.

**Comment by vuori** [comment]
Likely duplicate of #1030 ?

**Comment by pfrazee** [comment]
I was able to reproduce using the dat cli, but for some reason couldnt connect to any peers in Beaker.

At a glance, it really doesn't seem like this ought to have an issue, but it sure does. I'll try some debugging with the dat cli.

**Comment by pfrazee** [comment]
@mafintosh I dug around in chrome's inspector attached to the node process. What I saw was some kind of looping condition in `append-tree` that seems to be within a searching algorithm. I've attached a gif of what I saw. Note there seemed to be two loops; perhaps one metadata and one content?

![stack-execution](https://user-images.githubusercontent.com/1270099/42242763-a9838e0c-7ed4-11e8-86af-c6ec3381f46d.gif)
